artifacts.go

```diff
@@ -30,34 +33,63 @@ import (
 var Version = "unknown"
 
 func writeArtifact(c *cli.Context) error {
-	if len(c.StringSlice("device-type")) == 0 || len(c.String("artifact-name")) == 0 ||
+	if len(c.StringSlice("device-type")) == 0 ||
+		len(c.String("artifact-name")) == 0 ||
 		len(c.String("update")) == 0 {
 		return errors.New("must provide `device-type`, `artifact-name` and `update`")
 	}
 
-	he := &parser.HeaderElems{
-		Metadata: []byte(""),
+	// set default name
+	name := "artifact.mender"
+	if len(c.String("output-path")) > 0 {
+		name = c.String("output-path")
+	}
+	version := c.Int("version")
+	// set version to 2 if artifact is signed and version
+	// is not set explicitly
+	if !c.IsSet("version") && len(c.String("sign")) != 0 {
+		version = 2
```

Comment:
There is a flag you can use to set the version. What is more it is implemented so that default is 1 for unsigned artifact and 2 for signed. For now I think we can leave it like that. Later on we can deprecate 1 and use 2 as default.